core-api/src/main/kotlin/com/jakemoore/datakache/core/connections/DatabaseService.kt

```
@@ -374,4 +424,23 @@ internal abstract class DatabaseService : LoggerService, Service {
         eventHandler: ChangeEventHandler<K, D>
     ): ChangeStreamManager<K, D>
+
+    /**
+     * Override shutdown to clean up the update queue manager.
+     * Subclasses should call super.shutdown() in their implementation.
+     */
+    override suspend fun shutdown(): Boolean {
+        try {
+            updateQueueManager.shutdown()
+            return true
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)